Assets/Scripts/CameraFollow.cs

```diff
@@ -23,7 +25,9 @@ private void Start()
     {
         // Setting up the reference.
 		findPlayer();
```

Comment:
I think you can just add in maxSpeed = player.maxSpeed in findPlayer() method, that way u get the latest maxSpeed - one that might have been slowed down in controller2